server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -182,6 +193,20 @@ case object AsText extends ParseOption {
 case object AsAddresses extends ParseOption {
   override def extractHeaderValue(field: Field): Option[EmailHeaderValue] = Some(AddressesHeaderValue.from(field))
 }
+case object AsGroupedAddresses extends ParseOption {
+  override def extractHeaderValue(field: Field): Option[EmailHeaderValue] = Some(GroupedAddressesHeaderValue.from(field))
+}
+case object AsMessageIds extends ParseOption {
+  override def extractHeaderValue(field: Field): Option[EmailHeaderValue] = Some(MessageIdsHeaderValue.from(field))
+}
+case object AsDate extends ParseOption {
+  override def extractHeaderValue(field: Field): Option[EmailHeaderValue] = Some(DateHeaderValue.from(field, ZoneId.systemDefault()))
+
+  def extractHeaderValue(field: Field, zoneId: ZoneId): Option[EmailHeaderValue] = Some(DateHeaderValue.from(field, zoneId))
+}
+case object AsURLs extends ParseOption {
+  override def extractHeaderValue(field: Field): Option[EmailHeaderValue] = Some(URLsHeaderValue.from(field))
```

Comment:
why extractHeaderValue return an Option ? 
Because in all the implementations it return Some(...)